crates/rethnet_evm/benches/util.rs

```diff
@@ -1,75 +1,118 @@
+                    let number_of_accounts_per_checkpoint =
+                        number_of_accounts / number_of_checkpoints;
+                    for account_number in 1..=number_of_accounts_per_checkpoint {
+                        let account_number = (checkpoint_number
+                            * number_of_accounts_per_checkpoint)
+                            + account_number;
+                        let address = Address::from_low_u64_ne(account_number);
                         state
-                            .set_account_storage_slot(
+                            .insert_account(
                                 address,
-                                U256::from(storage_slot),
-                                U256::from(account_number),
+                                AccountInfo::new(
+                                    U256::from(account_number),
+                                    account_number,
+                                    Bytecode::new_raw(Bytes::copy_from_slice(address.as_bytes())),
+                                ),
                             )
                             .unwrap();
+                        for storage_slot in 0..number_of_storage_slots_per_account {
+                            state
+                                .set_account_storage_slot(
+                                    address,
+                                    U256::from(storage_slot),
+                                    U256::from(account_number),
+                                )
+                                .unwrap();
+                        }
                     }
+                    state.checkpoint().unwrap();
+                    checkpoints.push(state.state_root().unwrap());
                 }
-                state.checkpoint().unwrap();
+                snapshots.push(state.make_snapshot());
             }
         }
     }
 
     /// Returns a set of factories, each member of which produces a clone of one of the state objects in this struct.
-    fn make_clone_factories(
+    fn make_state_refs(
         &self,
     ) -> Vec<(
         &'static str, // label of the type of state produced by this factory
         Box<dyn Fn() -> Box<dyn SyncState<StateError>> + '_>,
+        &Vec<B256>,
+        &Vec<B256>,
     )> {
         vec![
-            ("LayeredState", Box::new(|| Box::new(self.layered.clone()))),
-            ("HybridState", Box::new(|| Box::new(self.hybrid.clone()))),
+            (
+                "Layered",
+                Box::new(|| Box::new(self.layered.clone())),
+                &self.layered_checkpoints,
+                &self.layered_snapshots,
+            ),
+            (
+                "Hybrid",
+                Box::new(|| Box::new(self.hybrid.clone())),
+                &self.hybrid_checkpoints,
+                &self.hybrid_snapshots,
+            ),
         ]
     }
 }
 
 #[cfg(feature = "bench-once")]
+#[allow(dead_code)]
```

Comment:
Why does this throw a dead code warning?

Reply:
it's because of the `SNAPSHOT_SCALES` member of this module, which is used in some of the benchmark binaries but not all. (each benchmark (eg `state_history` vs `state_debug`) is a separate binary and raises its own set of `dead_code` warnings/errors). i moved the `dead_code` directive, to apply specifically to `SNAPSHOT_SCALES` rather than this whole module, in https://github.com/NomicFoundation/hardhat/pull/3921/commits/776d41b48a0bb33af00592bcdb8996a7d9784a40